app/src/main/java/chat/rocket/android/chatroom/ui/PinnedMessagesAdapter.kt

```diff
@@ -0,0 +1,144 @@
+    }
+
+    override fun getItemId(position: Int): Long {
+        return dataSet[position].id.hashCode().toLong()
+    }
+
+    class ViewHolder(itemView: View,
+                     val serverUrl: String) : RecyclerView.ViewHolder(itemView) {
+
+        private lateinit var messageViewModel: MessageViewModel
+
+        fun bind(message: MessageViewModel) = with(itemView) {
+            messageViewModel = message
+            bindUserAvatar(message, image_avatar, image_unknown_avatar)
+            text_user_name.content = message.sender
+            text_message_time.content = message.time
+            text_content.content = message.content
+            text_content.movementMethod = LinkMovementMethod()
+
+            bindAttachment(message, message_attachment, image_attachment, audio_video_attachment,
+                    file_name)
+        }
+
+        private fun bindAttachment(message: MessageViewModel,
+                                   attachment_container: View,
+                                   image_attachment: SimpleDraweeView,
+                                   audio_video_attachment: View,
+                                   file_name: TextView) {
+            with(message) {
+                if (attachmentUrl == null || attachmentType == null) {
+                    attachment_container.setVisible(false)
+                    return
+                }
+
+                var imageVisible = false
+                var videoVisible = false
+
+                attachment_container.setVisible(true)
+                when (message.attachmentType) {
+                    is AttachmentType.Image -> {
+                        imageVisible = true
+                        image_attachment.setImageURI(message.attachmentUrl)
+                        image_attachment.setOnClickListener { view ->
+                            // TODO - implement a proper image viewer with a proper Transition
+                            ImageViewer.Builder(view.context, listOf(message.attachmentUrl))
+                                    .setStartPosition(0)
+                                    .show()
+                        }
+                    }
+                    is AttachmentType.Video,
+                    is AttachmentType.Audio -> {
+                        videoVisible = true
+                        audio_video_attachment.setOnClickListener { view ->
+                            message.attachmentUrl?.let { url ->
+                                PlayerActivity.play(view.context, url)
+                            }
+                        }
+                    }
+                }
+
+                image_attachment.setVisible(imageVisible)
+                audio_video_attachment.setVisible(videoVisible)
+                file_name.text = message.attachmentTitle
+            }
+        }
+
+        private fun bindUserAvatar(message: MessageViewModel, drawee: SimpleDraweeView, imageUnknownAvatar: ImageView) = message.getAvatarUrl(serverUrl).let {
```

Comment:
```kotlin
fun function() = statement()
```
works better for one liners... If it has multiple lines, lets use `{ }` and `return`